clients/src/main/java/org/apache/kafka/common/requests/ReadShareGroupStateSummaryResponse.java

```diff
@@ -64,4 +67,46 @@ public static ReadShareGroupStateSummaryResponse parse(ByteBuffer buffer, short
                 new ReadShareGroupStateSummaryResponseData(new ByteBufferAccessor(buffer), version)
         );
     }
+
+    public static ReadShareGroupStateSummaryResponseData toErrorResponseData(Uuid topicId, int partitionId, Errors error, String errorMessage) {
+        return new ReadShareGroupStateSummaryResponseData().setResults(
+                Collections.singletonList(new ReadShareGroupStateSummaryResponseData.ReadStateSummaryResult()
+                        .setTopicId(topicId)
+                        .setPartitions(Collections.singletonList(new ReadShareGroupStateSummaryResponseData.PartitionResult()
+                                .setPartition(partitionId)
+                                .setErrorCode(error.code())
+                                .setErrorMessage(errorMessage)))));
+    }
+
+    public static ReadShareGroupStateSummaryResponseData.PartitionResult toErrorResponsePartitionResult(int partitionId, Errors error, String errorMessage) {
+        return new ReadShareGroupStateSummaryResponseData.PartitionResult()
+                .setPartition(partitionId)
+                .setErrorCode(error.code())
+                .setErrorMessage(errorMessage);
+    }
+
+    public static ReadShareGroupStateSummaryResponseData toResponseData(
+            Uuid topicId,
+            int partition,
+            long startOffset,
+            int stateEpoch
+    ) {
+        return new ReadShareGroupStateSummaryResponseData()
+                .setResults(Collections.singletonList(
+                        new ReadShareGroupStateSummaryResponseData.ReadStateSummaryResult()
+                                .setTopicId(topicId)
+                                .setPartitions(Collections.singletonList(
```

Comment:
Because we've recently dropped Java 8, you can now use `List.of` instead of `Collections.singletonList`.